pkg/template/template.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package template
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+// TemplateData contains the data available for template substitution
+type TemplateData struct {
+	NodeName string `json:"nodeName"`
+}
+
+// ProcessParameterValue processes a parameter value string with Go template syntax
+// It replaces placeholders like {{.NodeName}} with actual values
+func ProcessParameterValue(paramValue string, nodeName string) (string, error) {
```

Comment:
nit: the function's name seems a bit too generic. Usually function handling templates are named like "RenderTemplate", we could call it "RenderParameterTemplate" for example.